examples/app_drive_mounted_s3/app.py

```diff
@@ -0,0 +1,165 @@
+            f.read()
+
+        print(f"Drive2(id={self.drive_2.id}) was mounted successfully!")
+
+        # -------- Done! --------
+        self.done = True
+
+
+class Work2(L.LightningWork):
+    def __init__(self, drive: "Drive", *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # you must assign the drive to an attribute on the work in the __init__ method
+        # in order for it to be mounted when running on the cloud runtime.
+        self.drive = drive
+
+        # signal to indicate if work has completed executing or not
+        self.done = None
+
+    def run(self, check_file: str):
+        print(f"verifying that {check_file} exists")
+        assert os.path.exists(check_file)
+
+        print(f"verifying that the byte size of {check_file} is reasonably reported & not just the inode metadata")
+        assert os.path.getsize(check_file) > 1_000  # bytes
+
+        print(f"verifying that we can open and read the contents of {check_file} without errors")
+        with open(check_file, "rb") as f:
+            f.read()
+
+        print("verifications complete! Exiting work run...")
+        self.done = True
+
+
+class Flow(L.LightningFlow):
+    def __init__(self):
+        super().__init__()
+
+        # Work1 contains Drive definitions in the __init__() method
+        self.work_1 = Work1(parallel=True)
+
+        # Work 2 will pass a drive we define in the flow into the work as an init argument
+        self.work2_drive_root_folder = "/tmp/drive1/datadir/"
+        # The directory must exist when initializing the flow and work, even if it is empty.
+        # We will create, and mount to this path automatically when `Work.run()` is called.
+        os.makedirs(self.work2_drive_root_folder, exist_ok=True)
+
+        # create the drive and initialize Work2 with the drive object
+        drive = Drive("s3://ryft-public-sample-data/twitter/", root_folder=self.work2_drive_root_folder)
+        self.work_2 = Work2(drive, parallel=True)
+
+        # we will call Work2.run() with a file path to check to ensure that it is available and mounted.
+        self.work2_drive_checkfile = os.path.join(self.work2_drive_root_folder, "twitter2015.tar.gz")
+
+        self.start_time = None
+
+    def run(self):
+        # set the start time only once for the Flow loop
+        if self.start_time is None:
+            self.start_time = time.time()
+
+        # Works which pass through the `parallel` argument to the L.Work base class run only once
+        # even though the main flow loop repeats infinitely if not explicitly exited.
+        self.work_1.run()
+        self.work_2.run(self.work2_drive_checkfile)
+
+        # are we done? If so, exit!
+        if self.work_1.done and self.work_2.done:
+            self._exit("Application End Successful!")
+
+        # If the works don't complete in < 30 minutes, a problem occured and we will exit here.
+        if time.time() - self.start_time > 30 * MINUTE:
```

Comment:
30 minutes are too long to run a simple two works with the default image, I would say if they take more than 5 minutes this is an issue.